src/main/java/com/google/devtools/build/remote/client/RemoteClient.java

```diff
@@ -376,29 +466,58 @@ private static void doCat(CatCommand options, RemoteClient client) throws IOExce
     }
   }
 
-  private static void doShowAction(ShowActionCommand options, RemoteClient client)
-      throws IOException {
-    Action.Builder builder = Action.newBuilder();
-    FileInputStream fin = new FileInputStream(options.file);
+  private static void doShowActionV1(File file, int limit, RemoteClient client) throws IOException {
```

Comment:
I think a better way to refactor this would be to have a getActionV1FromFile function, and use it both in doShowAction and doRun.